### PR TITLE
Update autotools, add Travis-CI support, and remove some unused code

### LINKED ISSUE #79: Update autotools
`pdsh` is long in need of an autotools update. Besides the likely hundereds of "deprecated" AC macro warnings, there are the following minor gotchas:
- Use of libtldl as a portable `dlopen()`. In the current code, and older libltdl is used that is a single drop-in `libltdl.[ch]` set of files. Newer libtoolize wants to copy an entire tree of files under `${top_srcdir}`/libltldl, along with its own autotools setup, etc.
- Currently products of ./bootstrap are committed to the repo. This made sense long ago, but now it is standard practice (and much cleaner) to not commit these products, so we should probably switch to that.

### PR DESCRIPTION
**Pull request by grondo**
The initial objective of these changes was to be able to update autotools for pdsh and remove autotools products from version control, in order to make addition of new modules much easier on the developer (e.g. see #91).

To make this easier, the use of libltdl was dropped in favor of straight `dlopen(3)` and `dlsym(3)`, which should be fine for most if not all pdsh users nowadays.

Along the way *many* modules that were presumed unused were dropped, including
 * Quadrics/Elan support (qshell, mqshell, and the rmsquery module)
 * The SDR module used to get lists of hosts from SDRGetObjects on IBM SPs
 * The `nodeattr` module (everyone uses libgenders now)

After this work, the `xpopen()` function had no users so it was removed.

The use of META and config/ac_meta.m4 were removed.

Finally, support for Travis-CI was added, and many compiler warnings were fixed along the way.

Support for `--enable-code-coverage` and `make check-code-coverage` was also added, but I couldn't get this working in Travis yet to send results to codecov.io.

@garlick, @chu11, happy to take any feedback here.

@tabaer, you might want to rebase #91 on this and see how it goes -- it should be much smoother.

Fixes #79

### CODE REVIEW

File: config/ac_pam.m4 [hunk 1]
```diff
@@ -21,8 +21,7 @@ AC_DEFUN([AC_PAM],
   #
   AC_MSG_CHECKING([for whether to build with pam support])
   AC_ARG_WITH([pam],
-    AC_HELP_STRING([--without-pam],
-      [Do not build qshell/mqshell with pam support]),
+    AS_HELP_STRING([--without-pam],[Do not build qshell/mqshell with pam support]),
     [ case "$withval" in
```
Comment (chu11) [inline config/ac_pam.m4]: we can probably remove ac_pam?  It appears to be specific to qshell/mqshell?

Comment (grondo) [inline config/ac_pam.m4]: Good catch forgot about that one.

**Comment by garlick** [comment]
There are a ton of great cleanups here.  Thanks for doing all this painful work!  Good riddance to the outdated modules too.

**Comment by garlick** [comment]
FWIW I had zero issues bootstrapping, building, and checking this branch from a freshly cloned repo on c9.io.

**Comment by grondo** [comment]
Thanks @garlick, @chu11. I'm going to merge this here and tag a pdsh-2.32. Then we should move the main project over to chaos/pdsh at some point.

**Pull request merged**